### PR TITLE
[Frontend][Tensorflow] dynamic support for StridedSlice

### PR DESCRIPTION
**Pull request by wangxiang2713**
Add dynamic support for TF StridedSlice op.
1. shape of input data may be dynamic.
2. if op output need reshape(e.g. shrink_axis_mask is set), target shape may be dynamic.

### CODE REVIEW

**Comment by tqchen** [comment]
cc @kevinthesun @mbrookhart please help to review this PR

File: python/tvm/relay/frontend/tensorflow.py [hunk 1]
```diff
@@ -44,6 +44,28 @@
 __all__ = ["from_tensorflow"]
 
 
+def check_symbolic_shape(shape):
+    return not all([isinstance(dim, (int, tvm.tir.IntImm)) for dim in shape])
```
Comment (mbrookhart) [inline python/tvm/relay/frontend/tensorflow.py:48]: There's a utility to do this here: https://github.com/apache/tvm/blob/e561007f0c330e3d14c2bc8a3ef40fb741db9004/python/tvm/relay/ty.py#L29

File: python/tvm/relay/frontend/tensorflow.py [hunk 2]
```diff
@@ -44,6 +44,28 @@
...
+
+
+def list_shape_of(tensor, ndim):
+    shape_tensor = _op.shape_of(tensor)
+    return [
+        _op.strided_slice(shape_tensor, begin=[i], end=[i + 1], strides=[1]) for i in range(ndim)
+    ]
```
Comment (mbrookhart) [inline python/tvm/relay/frontend/tensorflow.py:55]: I'm not sure I understand this utility, but perhaps `take` would be more appropriate than `strided_slice`?

**Review by mbrookhart — CHANGES_REQUESTED** [review]
Thanks for the contribution! Can you add a test that hits this?

**Comment by mbrookhart** [comment]
@wangxiang2713 Any chance you can get back to this?

**Comment by wangxiang2713** [comment]
I'm sorry that Tensorflow StridedSlice OP may meet many different situations for dynamic inputs, as input parameters begin_mask, end_mask, ellipsis_mask, new_axis_mask, shrink_axis_mask. We may try to solve this by change tensorflow graph. Thanks a lot for reviewing, close this.

**Pull request closed without merging**